app/src/main/java/nextstep/signup/ui/component/ValidatedTextField.kt

```diff
@@ -0,0 +1,54 @@
+package nextstep.signup.ui.component
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+
+@Composable
+fun ValidatedTextField(
+    field: InputFieldModel,
+    modifier: Modifier = Modifier,
+) {
+    fun validateField(value: String): Int? {
+        return if (value.isNotEmpty()) {
+            val result = field.validator.validate(value)
+            if (result.isValid.not()) result.message else null
+        } else {
+            null
+        }
+    }
+
+    val errorRes by remember(field.validator, field.value) {
+        mutableStateOf(validateField(field.value))
+    }
```

Comment:
지금 구조에서 Validator가 변경된 여부가 트리거되진 않아도 될 것 같아요! 예를 들어 다음과 같이 구현해보시는건 어떨까요?
```suggestion
    val errorRes by remember(value) {
        mutableStateOf(
            if (value.isNotEmpty()) {
                val result = validator.validate(value)
                if (result.isValid.not()) result.message else null
            } else {
                null
            }
        )
    }
```